type.go

```diff
@@ -297,6 +303,10 @@ func (t booleanType) AssignValue(dst reflect.Value, src Value) error {
 	return nil
 }
 
+func (t booleanType) ConvertValue(val Value, typ Type) (Value, error) {
+	return val, nil
```

Comment:
How about having a generic check that the kinds match at least for cases where we don't have conversion rules yet?

```suggestion
	return val, checkTypeKindEqual(t, typ)
```

```go
func checkTypeKindEqual(to, from Type) error {
	if to.Kind() != from.Kind() {
		return fmt.Errorf("cannot convert from parquet value of type %s to %s", from, to)
	}
	return nil
}
```

Reply:
That definitely seems useful. Updated in e516ee8